### PR TITLE
feat: Add exit template function

### PR DESCRIPTION
**Pull request by twpayne**
Refs #1670.

### CODE REVIEW

**Comment by twpayne** [comment]
I think the test failures are with Go 1.16, which doesn't seem to wrap template errors. The tests pass on Go 1.17.

As chezmoi supports the last two versions of Go (currently 1.16 and 1.17), we can only merge this once Go 1.18 is released. Converting to draft.

File: docs/REFERENCE.md [hunk 1]
```diff
@@ -2606,6 +2607,12 @@ used.
 
 ---
 
+### `stop`
+
+`stop` stops template execution, without returning an error.
```
Comment (felipecrs) [inline docs/REFERENCE.md:2612]: Ops, in fact I did need to error (making the chezmoi call return a non-zero code). Perhaps this function can accept a parameter like `exitCode`, which defaults to 0 and can be changed on a need basis?

**Review by felipecrs — COMMENTED** [review]
Sounds amazing, except that for my use case I would also need to make chezmoi to exit with a non-zero code. I suggested one approach.

File: docs/REFERENCE.md (hunk 1 above)
Comment (twpayne) [reply docs/REFERENCE.md]: OK, PR updated. PTAL.

**Comment by felipecrs** [comment]
This is brilliant! Just what I needed, thank you so much!

**Comment by felipecrs** [comment]
Regarding Go 1.16, just for curiosity, is there any place using it to build chezmoi? Like external package managers or something like that? It's really just for curiosity, I'm not asking to change anything.

**Comment by twpayne** [comment]
> Regarding Go 1.16, just for curiosity, is there any place using it to build chezmoi?

Yeah, from memory Arch Linux and Illumnos are having difficulty moving on from Go 1.16.

**Comment by felipecrs** [comment]
> Yeah, from memory Arch Linux and Illumnos are having difficulty moving on from Go 1.16.

Understood. Thank you.

**Comment by twpayne** [comment]
I've updated the code to only include the `exit` template function when chezmoi is built with Go 1.17 or later so this can be merged now.

**Comment by felipecrs** [comment]
That's awesome, thanks a lot!

**Pull request merged**